src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/PandocCompiler.kt

```diff
@@ -0,0 +1,47 @@
+/**
+    Copyright 2018 BarD Software s.r.o
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Author: Mikhail Shavkunov (@shavkunov)
+ */
+
+package com.bardsoftware.backend.fleet.rmarkdown
+
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+import org.apache.commons.lang.text.StrSubstitutor
+import org.slf4j.LoggerFactory
+
+private val LOGGER = LoggerFactory.getLogger("Pandoc")
+const val PANDOC_DEFAULT_FONT = "DejaVu Sans"
+val DEFAULT_CONFIG = ConfigFactory.load()
+val pandocArguments = listOf("projectRootAbsPath", "workingDirRelPath",
+        "inputFileName", "outputFileName", "mainFont")
+
+fun compile(config: Config, substituteArguments: List<String> = pandocArguments, vararg args: String) {
```

Comment:
I recommend using data class for the arguments passing. Because you really don't want unknown argument names and you really don't want mismatching or misordered argument values. You can convert data class to map (for string substitution purposes) using either ObjectMapper or by [delegating storage to map](https://kotlinlang.org/docs/reference/delegated-properties.html#storing-properties-in-a-map)